src/stampy_nlp/search.py

```diff
@@ -27,7 +28,17 @@ def format_matches(results):
     return results_list
 
 
-def semantic_search(query: str, top_k: int = DEFAULT_TOPK, showLive: bool = True, status=[], namespace: str = 'faq-titles'):
+def add_doc_content(results):
+    def update(result):
+        result['content'] = get_contents(result['pageid'])
+        return result
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
```

Comment:
it turns out that Coda doesn't have a nice way to filter rows :/ You can either fetch them one by one, or loop over the whole table. Both would involve multiple calls, this way may hopefully be faster?